astropy/io/misc/asdf/tags/table/tests/test_table.py

```diff
@@ -98,8 +98,15 @@ def test_table_inline(tmpdir):
     def check(ff):
         assert len(list(ff.blocks.internal_blocks)) == 0
 
-    helpers.assert_roundtrip_tree({'table': t}, tmpdir, asdf_check_func=check,
-                                  write_options={'auto_inline': 64})
+    if hasattr(asdf, 'config_context'):
```

Comment:
Why not use version check? That way, we can easily search for dead code when we bump:

https://github.com/astropy/astropy/blob/715d7a660a554f8880d8df82fae28180eeb21e2f/setup.cfg#L80

Also see: https://github.com/astropy/astropy/blob/main/astropy/utils/compat/optional_deps.py